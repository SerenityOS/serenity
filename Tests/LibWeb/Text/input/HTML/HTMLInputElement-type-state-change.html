<!DOCTYPE html>
<script src="../include.js"></script>
<script>
    test(() => {
        const testData = [
            { initialType: "text", newType: "email", initialValue: "  test  " },
            { initialType: "text", newType: "URL", initialValue: "  test  " },
            { initialType: "hidden", newType: "text", initialValue: "  t\re\nst  " },
            { initialType: "hidden", newType: "search", initialValue: "  t\re\nst  " },
            { initialType: "hidden", newType: "tel", initialValue: "  t\re\nst  " },
            { initialType: "hidden", newType: "password", initialValue: "  t\re\nst  " },
            { initialType: "hidden", newType: "email", initialValue: "  t\re\nst  " },
            { initialType: "hidden", newType: "URL", initialValue: "   t\re\nst  " },
            { initialType: "text", newType: "number", initialValue: "123.45" },
            { initialType: "text", newType: "number", initialValue: "not-a-number" },
            { initialType: "text", newType: "date", initialValue: "not-a-date" },
            { initialType: "text", newType: "date", initialValue: "2024-09-10" },
            { initialType: "text", newType: "date", initialValue: "2024--09--10" },
            { initialType: "text", newType: "datetime-local", initialValue: "not-a-local-datetime" },
            { initialType: "text", newType: "datetime-local", initialValue: "2024-09-10 23:59" },
            { initialType: "text", newType: "datetime-local", initialValue: "2024-09-10T23:59" },
            { initialType: "text", newType: "datetime-local", initialValue: "2024--09--10TT23::59" },
            { initialType: "text", newType: "month", initialValue: "not-a-month" },
            { initialType: "text", newType: "month", initialValue: "2024-09" },
            { initialType: "text", newType: "month", initialValue: "2024--09" },
            { initialType: "text", newType: "week", initialValue: "not-a-week" },
            { initialType: "text", newType: "week", initialValue: "2024-W26" },
            { initialType: "text", newType: "week", initialValue: "2024--W26" },
            { initialType: "text", newType: "time", initialValue: "not-a-time" },
            { initialType: "text", newType: "time", initialValue: "23:59:59" },
            { initialType: "text", newType: "time", initialValue: "23::59::59" },
        ];
        for (const { initialType, newType, initialValue } of testData) {
            const inputElement = document.createElement("input");
            inputElement.type = initialType;
            inputElement.value = initialValue;
            inputElement.type = newType;
            const initialValueForDisplay = initialValue.replace("\r", "\\r").replace("\n", "\\n");
            const mutatedValue = inputElement.value.replace("\r", "\\r").replace("\n", "\\n");
            println(`Changing type from "${initialType}" to "${newType}" changes value from "${initialValueForDisplay}" to "${mutatedValue}"`);
        }
    });
</script>
