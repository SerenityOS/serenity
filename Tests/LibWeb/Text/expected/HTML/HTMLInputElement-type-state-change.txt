Changing type from "text" to "email" changes value from "  test  " to "test"
Changing type from "text" to "URL" changes value from "  test  " to "test"
Changing type from "hidden" to "text" changes value from "  t\re\nst  " to "  test  "
Changing type from "hidden" to "search" changes value from "  t\re\nst  " to "  test  "
Changing type from "hidden" to "tel" changes value from "  t\re\nst  " to "  test  "
Changing type from "hidden" to "password" changes value from "  t\re\nst  " to "  test  "
Changing type from "hidden" to "email" changes value from "  t\re\nst  " to "test"
Changing type from "hidden" to "URL" changes value from "   t\re\nst  " to "test"
Changing type from "text" to "number" changes value from "123.45" to "123.45"
Changing type from "text" to "number" changes value from "not-a-number" to ""
Changing type from "text" to "date" changes value from "not-a-date" to ""
Changing type from "text" to "date" changes value from "2024-09-10" to "2024-09-10"
Changing type from "text" to "date" changes value from "2024--09--10" to ""
Changing type from "text" to "datetime-local" changes value from "not-a-local-datetime" to ""
Changing type from "text" to "datetime-local" changes value from "2024-09-10 23:59" to "2024-09-10T23:59"
Changing type from "text" to "datetime-local" changes value from "2024-09-10T23:59" to "2024-09-10T23:59"
Changing type from "text" to "datetime-local" changes value from "2024--09--10TT23::59" to ""
Changing type from "text" to "month" changes value from "not-a-month" to ""
Changing type from "text" to "month" changes value from "2024-09" to "2024-09"
Changing type from "text" to "month" changes value from "2024--09" to ""
Changing type from "text" to "week" changes value from "not-a-week" to ""
Changing type from "text" to "week" changes value from "2024-W26" to "2024-W26"
Changing type from "text" to "week" changes value from "2024--W26" to ""
Changing type from "text" to "time" changes value from "not-a-time" to ""
Changing type from "text" to "time" changes value from "23:59:59" to "23:59:59"
Changing type from "text" to "time" changes value from "23::59::59" to ""
